learning/01_Exercises/Lab 7 to 9/UI/userInterface.py

```diff
@@ -0,0 +1,486 @@
+            self.refreshScreen()
+
+            # Display the title and the current list with clients.
+            print("REMOVE CLIENTS")
+            self.displayClients()
+
+            # Collect the movie ID.
+            ID = self.collectID()
+
+            # Try/remove the movie with the ID given.
+            self.controller.removeClient(ID)
+
+            # Ask the user if he wants to continue
+            if not self.exit():
+                break
+
+    def updateMovies(self):
+        while True:
+            # Refresh the screen.
+            self.refreshScreen()
+
+            # Display the title and the current list with clients.
+            print("UPDATE MOVIES")
+            self.displayMovies()
+
+            # Collect the movie ID.
+            ID = self.collectID()
+            if not self.controller.validatorForMovie.checkID(ID, self.controller.moviesRepository.getIdList()):
+                # You must raise an error/message for the user to know what's going on.
+                continue
+
+            # Collect new data to update the list.
+            newMovie = self.collectMovieData()
+            if not self.controller.validatorForMovie.validator(newMovie, self.controller.moviesRepository.getNames()):
+                # You must raise an error/message for the user to know what's going on.
+                continue
+
+            # Try to update by ID.
+            self.controller.updateMovie(ID, newMovie)
+
+            # Ask the user if he wants to continue
+            if not self.exit():
+                break
+
+    def updateClients(self):
+        while True:
+            # Refresh the screen.
+            self.refreshScreen()
+
+            # Display the title and the current list with clients.
+            print("UPDATE CLIENTS")
+            self.displayClients()
+
+            # Collect the client's ID.
+            ID = self.collectID()
+            if not self.controller.validatorForClient.checkID(ID, self.controller.clientsRepository.getIdList()):
+                # You must raise an error/message for the user to know what's going on.
+                continue
+
+            # Collect new data to update the list.
+            newClient = self.collectClientsData()
+            if not self.controller.validatorForClient.validator(newClient, self.controller.clientsRepository.getNames()):
+                # You must raise an error/message for the user to know what's going on.
+                continue
+
+            # Try to update by ID.
+            self.controller.updateClient(ID, newClient)
+
+            # Ask the user if he wants to continue
+            if not self.exit():
+                break
+
+    def searchMovies(self):
+        while True:
+            # Refresh the screen.
+            self.refreshScreen()
+
+            # Display the title.
+            print("SEARCH FOR MOVIES")
+
+            # Collect the name of the movie.
+            findMovie = self.collectWord()
```

Comment:
You can have a better name for this var like `search_term`

Reply:
Rezolvat.